### PR TITLE
Solution 2: Repeat characters

### PR DESCRIPTION
**Pull request by Papsik09**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by Mykytako — DISMISSED** [review]
Passed all the checks, nice job!

File: solutions/tests/test_repeat_characters.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        self.assertEqual(repeat_characters("loop"), "lloooopp")
+
+    def test_upper_case(self):
+        """Test if will mulltiply upper characters"""
+        self.assertEqual(repeat_characters("BOOM"), "BBOOOOMM")
+
+    def test_upper_lower_case(self):
+        """Test if will mulltiply upper and lower characters"""
+        self.assertEqual(repeat_characters("MoThEr"), "MMooTThhEErr")
+
```
Comment (olumide-AI) [inline solutions/tests/test_repeat_characters.py:35]: - I suggest you should add more test (edge cases and defensive test) for example testing: empty strings, strings with spaces, strings with numbers, test NONE etc. 
-

File: solutions/repeat_characters.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+1/4/2025
+
+author by: Dmytro Klymenko
+
+"""
+
+
+def repeat_characters(word: str) -> str:
+    """
+    Function repeat all characters in word 2 times and return long word
```
Comment (olumide-AI) [inline solutions/repeat_characters.py:13]: You can use a more descriptive line ex;
-Repeats each character in the input string twice and returns the resulting string.

File: solutions/repeat_characters.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      Args:
+        repeat_characters(str) - desired word
```
Comment (olumide-AI) [inline solutions/repeat_characters.py:15]: You can use a more descriptive line like;
- word (str): The input string whose characters are to be repeated.

File: solutions/repeat_characters.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    Returns:
+        str: Desired word with doubled characters
```
Comment (olumide-AI) [inline solutions/repeat_characters.py:18]: I suggest refactoring line like this
-str: A new string where each character from the input is repeated twice.

File: solutions/repeat_characters.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    Examples:
```
Comment (olumide-AI) [inline solutions/repeat_characters.py:20]: I believe you shouldn't use print in your doc string, you also forgot the quotation marks on the return output
I suggest you try it this way:
Examples:
    >>> repeat_characters("Winter")
    'WWiinntteerr'
    >>> repeat_characters("peach")
    'ppeeaacchh'
    >>> repeat_characters("cola")
    'ccoollaa'

**Review by olumide-AI — COMMENTED** [review]
Great job with your code, i added some single line comments you could look at, where i gave some suggestions. Once you resolve the comments let me know and i will approve solution. Thank you

**Review by Mykytako — APPROVED** [review]

**Review by olumide-AI — APPROVED** [review]
Looks, good. Thanks for making the necessary changes
